packages/offix-client/src/conflicts/ConflictLink.ts

```diff
@@ -0,0 +1,178 @@
+import { onError, ErrorResponse } from "apollo-link-error";
+import { GraphQLError, ExecutionResult, subscribe } from "graphql";
+import { ApolloLink, Operation, NextLink, Observable, FetchResult } from "apollo-link";
+import { ConflictResolutionData } from "./strategies/ConflictResolutionData";
+import { isMutation } from "../utils/helpers";
+import { ObjectState, ConflictListener } from ".";
+import { ConflictResolutionStrategy } from "./strategies/ConflictResolutionStrategy";
+import { clientWins } from "./strategies/strategies";
+import { ConflictHandler } from "./handler/ConflictHandler";
+
+/**
+ * Local conflict thrown when data outdates even before sending it to the server.
+ * Can be used to correct any data in flight or shown user another UI to visualize new state
+ */
+export class LocalConflictError extends Error {
+  constructor(private base: any, private variables: any) {
+    super();
+  }
+}
+
+/**
+ * Represents conflict information that was returned from server
+ */
+export interface ConflictInfo {
```

Comment:
Potential Overlap with the interface defined here? https://github.com/aerogear/offix/pull/54/files#diff-b8f9c4f82651357a985e9cfd8207cd4eR9

Should we deduplicate these things into a commmon package at some point such that we guarantee any interfaces share between client and server match up? I've been burned by this before. Defining the same interface in two different places. Had to make changes later and ended up with runtime errors that were not caught at compile time. Not saying let's do it right now but it's food for thought!